File: examples/rust-sdk/src/provenance_spec.rs
//! The spec declares traceability; production code stays unaware.

use provenance_sdk::{implemented_by, provenance_spec, requirement, rule, source};

provenance_spec!(pub share_links => "share-links" {
    requirement("sharing")
        .statement("Users can securely share documentation")
        .from(source("sharing-policy").document("docs/sharing-policy.md"))
        .rules([implemented_by!(
            rule("expiry").statement("Share links must expire within 30 days"),
            "src/share_links.rs",
            create_share_link
        )]),
});

File: examples/typescript-sdk/src/provenance.spec.ts
import { defineSpec, requirement, rule, source } from "@quality-sh/provenance";

import { createShareLink } from "./share-links.js";

export const shareLinks = defineSpec("share-links")
  .requirements(
    requirement("sharing")
      .statement("Users can share the documents")
      .from(source("sharing-policy").document("docs/sharing-policy.md"))
      .rules(
        rule("expiry")
          .statement("Share links must expire in 30 days")
          .implementedBy(createShareLink),
      ),
  )
  .build();
